src/test/java/uk/gov/hmcts/reform/draftstore/service/mappers/ToDbModelMapperTest.java

```diff
@@ -12,20 +14,33 @@
 
     @Test
     public void should_set_only_encrypted_document_when_secret_is_passed() throws Exception {
-        result = ToDbModelMapper.toDb(createDraft(), "a98sdf8asd7f");
+        result = ToDbModelMapper.toDb(createDraft(), new Secrets(SampleSecret.get(), null));
 
         assertThat(result.document).isNull();
         assertThat(result.encryptedDocument).isNotNull();
     }
 
     @Test
     public void should_set_only_unencrypted_document_when_secret_is_NOT_passed() throws Exception {
-        result = ToDbModelMapper.toDb(createDraft(), null);
+        result = ToDbModelMapper.toDb(createDraft(), new Secrets(null, null));
 
         assertThat(result.document).isNotNull();
         assertThat(result.encryptedDocument).isNull();
     }
 
+    @Test
+    public void should_use_primary_secret_for_encryption() throws Exception {
```

Comment:
Now that we may have two keys, we should check if:

 - the mapper will use the secondary key (with success) when the primary one fails
 - if the mapper will fail in the expected way when the secondary key fails

Reply:
done.